metagpt/actions/write_plan.py

```diff
@@ -0,0 +1,104 @@
+# -*- encoding: utf-8 -*-
+"""
+@Date    :   2023/11/20 11:24:03
+@Author  :   orange-crow
+@File    :   plan.py
+"""
+import json
+from copy import deepcopy
+from typing import Dict, List, Tuple
+
+from metagpt.actions import Action
+from metagpt.logs import logger
+from metagpt.prompts.ml_engineer import ASSIGN_TASK_TYPE_CONFIG, ASSIGN_TASK_TYPE_PROMPT
+from metagpt.schema import Message, Plan, Task
+from metagpt.utils.common import CodeParser, create_func_config
+
+
+class WritePlan(Action):
```

Comment:
use ActionNode

Reply:
will be restructured after new ActionNode